internal/project/documentregistry.go

```diff
@@ -118,6 +119,14 @@ func (r *DocumentRegistry) getDocumentWorker(
 	}
 }
 
+func (r *DocumentRegistry) getFileVersion(file *ast.SourceFile, options *core.CompilerOptions) int {
+	key := newRegistryKey(options, file.Path(), file.ScriptKind)
+	if entry, ok := r.documents.Load(key); ok {
+		return entry.version
```

Comment:
This is not enough i think. The source file may have been updated for the compilerOptions by other program and hence the source file may be newer = script info version. You actually want to check if `file == entry.sourceFile` as well